atlassian/rest_client.py

```diff
@@ -221,17 +265,36 @@ def request(
             json_dump = None if not json else dumps(json)
         self.log_curl_debug(method=method, url=url, headers=headers, data=data if data else json_dump)
         headers = headers or self.default_headers
-        response = self._session.request(
-            method=method,
-            url=url,
-            headers=headers,
-            data=data,
-            json=json,
-            timeout=self.timeout,
-            verify=self.verify_ssl,
-            files=files,
-            proxies=self.proxies,
-        )
+
+        backoff = 1
+        retries = 0
+        responseloop = True
+        while responseloop:
+            response = self._session.request(
+                method=method,
+                url=url,
+                headers=headers,
+                data=data,
+                json=json,
+                timeout=self.timeout,
+                verify=self.verify_ssl,
+                files=files,
+                proxies=self.proxies,
+            )
+            responseloop = False
```

Comment:
Setting always to `False` and in Line 295 to `True` is confusing. Remove line 284 and 295.

Reply:
@Spacetown I think that would cause an infinite loop if `self.backoffandretry` is set to `False`, which is the default.

If it's confusing is there another way we could implement it?

Now that you mention it I'm also just looking at line 288 and I think that could probably be removed although I think I put that in there to future proof any other logic changes relating to `responseloop` that might change the default - ie. making the logic explicit - so would probably vote to leave that in.